exonum/src/runtime/mod.rs

```diff
@@ -761,6 +768,37 @@ impl<'a> ExecutionContext<'a> {
             .initiate_adding_service(spec)
             .map_err(From::from)
     }
+
+    /// TODO Documentation ECR-3773
+    pub(crate) fn initiate_resuming_service(
+        &mut self,
+        instance_id: InstanceId,
+        params: impl BinaryValue,
+    ) -> Result<(), ExecutionError> {
+        let spec = DispatcherSchema::new(&*self.fork)
+            .get_instance(instance_id)
+            .ok_or(CoreError::IncorrectInstanceId)?
+            .spec;
+
+        let runtime = self
+            .dispatcher
+            .runtime_by_id(spec.artifact.runtime_id)
+            .ok_or(CoreError::IncorrectRuntime)?;
+        runtime
+            .initiate_resuming_service(self.reborrow(), &spec, params.into_bytes())
+            .map_err(|mut err| {
+                err.set_runtime_id(spec.artifact.runtime_id)
+                    .set_call_site(|| CallSite {
+                        instance_id,
+                        call_type: CallType::Constructor,
+                    });
+                err
+            })?;
+
+        DispatcherSchema::new(&*self.fork)
```

Comment:
Nit: Dispatcher schema is created twice in function, maybe worth storing it in the variable.

Reply:
I cannot store schema in variable because borrow checker against me in this case.